cranelift/filetests/filetests/isa/riscv64/icmp-imm-lhs.clif

```diff
@@ -0,0 +1,26 @@
+test compile precise-output
+set opt_level=speed
+target riscv64
+
+;; We don't have specific rules for iconst on the lhs of a icmp.
+;; Instead we rely on the mid end rewrite rules to push it to the rhs.
+;; This tests ensures that that works with optimizations enabled.
```

Comment:
Just to check: If the midend rules to flip constants to the right-hand side don't fire, perhaps because the egraphs pass doesn't get run, after this PR the backend will still produce correct code, right? It just won't be as efficient as it could be?

Reply:
Yes, exactly! Most of our runtests run in that mode. However I don't know how many have LHS constants.

Reply:
While looking today at other discussions about commutativity, I remembered that I don't expect this to be reliable even when the egraph pass runs.

Those mid-end rules declare that both commutative alternatives are equivalent. But the cost function will assign the same cost to both choices, since they're the same opcode and use the same operands.

When two choices have equal costs I can't guarantee anything about which one the egraph extraction phase will choose and send on to the backend for lowering.

Reply:
Makes sense, I'll add both versions of this rule. Thanks for looking into it!